.github/workflows/gen_doc.yml

```
@@ -0,0 +1,46 @@
+name: Generate_Doc
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    name: ${{ matrix.os }}, py-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
+        python-version: ["3.6"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - uses: ammaraskar/sphinx-action@master
+        with:
+          docs-folder: "docs/"
```

This sphinx-action is a bit overkill. Just add the dependencies to the dev dependencies in `pyproject.toml` instead (edit: they're already there) and do the same as we do in eeg-notebooks (but without the `pip install -r requirements.txt`): https://github.com/NeuroTechX/eeg-notebooks/pull/24/files

```suggestion
      - name: Install dependencies
        run: |
          pip install poetry
          poetry install
      - name: Build docs
        run: |
          cd docs && make html
```